karma.middleware.js

```diff
@@ -17,12 +17,16 @@ module.exports = function() {
       }
 
       let responseText;
-
       switch (request._parsedUrl.pathname) {
         case '/json':
```

Comment:
Instead of using a try...catch, add another case, e.g. `/postJSON`. Then the `postJSON` request can send to that path and get back the response it is expecting.